apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerReporter.java

```diff
@@ -155,83 +170,80 @@ public long getReported() {
         return reportingEventHandler.getReported();
     }
 
-    /**
-     * Flushes pending {@link ErrorCapture}s and {@link Transaction}s to the APM server.
-     * <p>
-     * This method may block for a while until a slot in the ring buffer becomes available.
-     * </p>
-     *
-     * @return A {@link Future} which resolves when the flush has been executed.
-     */
     @Override
     public Future<Void> flush() {
-        final boolean success = disruptor.getRingBuffer().tryPublishEvent(FLUSH_EVENT_TRANSLATOR);
+        CompletableVoidFuture future = new CompletableVoidFuture();
```

Comment:
Since we now have a mechanism to wait for an event to be processed based on the sequence, the only benefit of the returned `Future` is the ability to cancel. If we decide that cancelling is not that important, we can considerably simplify things (no need for the Java-7-compatible `CompletableFuture` and no need to use event translators that get `null` futures.

Reply:
We do need the canceling for [`api_request_time`](https://www.elastic.co/guide/en/apm/agent/java/current/config-reporter.html#config-api-request-time).
If we end the request due to `api_request_size` before `api_request_time` elapses, we need to cancel the `timeoutTask` which, in turn, cancels the future.